larray/inout/xw_reporting.py

```diff
@@ -0,0 +1,709 @@
+    from larray.inout.xw_excel import open_excel
+
+    class ItemSize(object):
+        def __init__(self, width, height):
+            self.width = width
+            self.height = height
+
+        @property
+        def width(self):
+            return self._width
+
+        @width.setter
+        def width(self, width):
+            _positive_integer(width)
+            self._width = width
+
+        @property
+        def height(self):
+            return self._height
+
+        @height.setter
+        def height(self, height):
+            _positive_integer(height)
+            self._height = height
+
+    class AbstractExcelItem(ItemSize):
+
+        _default_size = ItemSize(1, 1)
+
+        def __init__(self, top, left, width, height):
+            ItemSize.__init__(self, width, height)
+            self.top = top
+            self.left = left
+
+        def dump(self, sheet, data_sheet, row):
+            raise NotImplementedError()
+
+    class ExcelTitleItem(AbstractExcelItem):
+
+        _default_size = ItemSize(1000, 50)
+
+        def __init__(self, text, fontsize, top, left, width, height):
+            AbstractExcelItem.__init__(self, top, left, width, height)
+            self.text = str(text)
+            _positive_integer(fontsize)
+            self.fontsize = fontsize
+
+        def dump(self, sheet, data_sheet, row):
+            data_cells = data_sheet.Cells
+            # add title in data sheet
+            data_cells(row, 1).Value = self.text
+            # generate title banner in destination sheet
+            msoShapeRectangle = 1
+            msoThemeColorBackground1 = 14
+            sheet_shapes = sheet.Shapes
+            shp = sheet_shapes.AddShape(Type=msoShapeRectangle, Left=self.left, Top=self.top,
+                                        Width=self.width, Height=self.height)
+            fill = shp.Fill
+            fill.ForeColor.ObjectThemeColor = msoThemeColorBackground1
+            fill.Solid()
+            shp.Line.Visible = False
+            frame = shp.TextFrame
+            chars = frame.Characters()
+            chars.Text = self.text
+            font = chars.Font
+            font.Color = 1
+            font.Bold = True
+            font.Size = self.fontsize
+            frame.HorizontalAlignment = HAlign.xlHAlignLeft
+            frame.VerticalAlignment = VAlign.xlVAlignCenter
+            shp.SetShapesDefaultProperties()
+            # update and return current row position
+            return row + 2
+
+    _default_items_size['title'] = ExcelTitleItem._default_size
+
+    # TODO: fix problem with nans values
+    class ExcelGraphItem(AbstractExcelItem):
+
+        _default_size = ItemSize(427, 230)
+
+        def __init__(self, title, data, template, top, left, width, height):
+            AbstractExcelItem.__init__(self, top, left, width, height)
+            self.title = str(title)
+            data = aslarray(data)
+            if not 0 < data.ndim <= 2:
```

Comment:
this might be only me, but I always prefer when "chained comparisons" like this use the same operator on both sides. I don't know exactly why but I can "parse" them much quicker. Eg. `if not (1 <= data.ndim <= 2):`